pulser-simulation/pulser_simulation/hamiltonian.py

```diff
@@ -351,6 +353,12 @@ def _construct_hamiltonian(self, update: bool = True) -> None:
         Also builds qutip.Qobjs related to the Sequence if not built already,
         and refreshes potential noise parameters by drawing new at random.
 
+        Warning:
+            The refreshed noise parameters (when update=True) are only those
+            that change from shot to shot (ie doppler and state preparation).
+            Amplitude fluctuations change from pulse to pulse and are always
```

Comment:
I don't find the expression "pulse to pulse" very clear, what do you mean by this ? Aren't amplitude fluctuations updated whatever the value of update ?

Reply:
I mean that from one pulse to another, the amplitude fluctuation is always updated. How would it be clearer for you?

Reply:
Hum I am puzzled, is it really between two Pulses ? Isn't it for a whole SequenceSamples ?

Reply:
Yes, check it out here: https://github.com/pasqal-io/Pulser/blob/e16257ad5f9c6a88cbf3ffffb7f36aa6b2db129d/pulser-simulation/pulser_simulation/hamiltonian.py#L201-L233

`add_noise()` is applied on each pulse time slot and every time it is called, the random distribution is sampled. Contrast this against the doppler detuning which is stored in `self._doppler_detune` and is only updated when `_update_noise()` is called, ie between runs

Reply:
The idea is that the atoms are slow compared to the timescale of a shot, so you can assume the atoms stay in the same place on a single run. On the other hand, the amplitude fluctuations are fast, so they change within a run

Reply:
The idea is that the atoms are slow compared to the timescale of a shot, so you can assume the atoms move in the same direction with constant speed over a single run (and hence the detuning from the doppler effect stays constant over the run). On the other hand, the amplitude fluctuations are fast, so they change within a run

Reply:
Ok, I hadn't see that, thanks for the explanation ! Now that I see it, the implementation is perhaps a bit weird, since in the case of a Sequence containing only one super-long pulse, the value of noise_base that will be applied will be the same all along - there will be no "fluctuation" actually.
I would suggest to define a rate at which fluctuations should happen (and perhaps they should fluctuate at each ns ?), and sample them at each run (as it's done at the moment).

Reply:
I agree but it's out of scope, could you open an issue please?